### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by 78fatemanazari**
### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct [GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### HTML/CSS requirements

- Follow our list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

### JavaScript requirements

- Follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements

- Add a new JavaScript file and import it as a module:    
    - it will contain methods related to the status updates (`completed`: `true` / `false`).
- Add event listener to the checkbox (`change`).
- Update items object's value for `completed` key upon user actions.
- Implement a function for the "Clear all completed" button (use `filter()` method).
- Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

File: src/statusUpdate.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  tasksLocal[taskIndex - 1].completed = status;
+  localStorage.setItem('tasks', JSON.stringify(tasksLocal));
+}
+
+// This function clears all completed tasks from the list
+function clearCompletedTasks(tasksLocal) {
+  tasksLocal = tasksLocal.filter((task) => !task.completed);
+  localStorage.setItem('tasks', JSON.stringify(tasksLocal));
+  document.location.reload();
+}
```
Comment (souad988) [inline src/statusUpdate.js:12]: Good job so far 👍🏼  , but you still need to update indexes after removing completed tasks as not doing so is introducing bags when trying to updateTaskStatus 

Capture d’écran du 2023-07-28 11-38-12

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @78fatemanazari , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

242722065-db428b0c-8dfc-49c4-a156-42f227e32868


## Highlights
Export and import functionalities from modules
Update localStorage on state change


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/statusUpdate.js (hunk 1 above)
Comment (78fatemanazari) [reply src/statusUpdate.js:12]: Hi dear @souad988 
I don't have any error while editing the tasks.
Capture

Comment (souad988) [reply src/statusUpdate.js:12]: hi @78fatemanazari  , the problem is after you clear completed tasks if the list has [1,2,3,4,5] as item indexes and 1 3 4 are completed tasks then after clearCompletedTaskes the list will be  [2,5]  when we try to check task 2 as completed edit function looks for tasks[index - 1] => tasks[1] while it index 1 doesn't exists so we need to update indexes after clearCompletedTaskes

**Review by BregornOriginal — APPROVED** [review]
Hi @78fatemanazari 

Your project is complete!  There is nothing else to say other than... it's time to merge it 

Congratulations! 

giphy (7)

## Highlights

✅ Linters are passing
✅ Good work implementing indexes for each task
✅ Excellent commit history

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**